backend/test_direct_action_tool_calling.py

```
@@ -0,0 +1,502 @@
+#!/usr/bin/env python3
+"""
+Test file for direct action tool calling functionality.
+Tests the updated handle_direct_action method with actual tool execution.
+"""
+
+import asyncio
+import json
+import sys
+import os
+from unittest.mock import Mock, AsyncMock, patch
```

Ruff (0.12.2)

`unittest.mock.patch` imported but unused

Remove unused import: `unittest.mock.patch`

(F401)